src/server/main_service.cc

```diff
@@ -1590,8 +1572,27 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  bool scheduled =
-      StartMultiExec(cntx->db_index(), cntx->transaction, &exec_info, &tmp_keys, global_script);
+  bool transactional = global_script;
```

Comment:
Why transactional is set to global_script and not to false?

Reply:
if global_script is true - transactional is true for sure. Otherwise it's false.